src/service/firmware-update.toit

```diff
@@ -189,12 +189,16 @@ class FirmwarePatcher_ implements PatchObserver:
 
   apply_ reader/io.Reader offset/int old-mapping/firmware.FirmwareMapping? -> none:
     binary-patcher := Patcher reader old-mapping --patch-offset=offset
-    if not binary-patcher.patch this:
-      // This should only happen if we to get the wrong bits
-      // served to us. It is unlikely, but we log it and throw
-      // an exception so we can try to recover.
-      logger_.error "firmware update: failed to apply patch"
-      throw "INVALID_FORMAT"
+    try:
```

Comment:
I'm not entirely sure why, but I don't like this too much. I'd prefer the 'processed' count to be a tag for an existing error log entry, so maybe something like this would be better.

```
binary-patcher := Patcher reader old-mapping --patch-offset=offset
    try:
      if not binary-patcher.patch this:
        // This should only happen if we to get the wrong bits
        // served to us. It is unlikely, but we log it and throw
        // an exception so we can try to recover.
        throw "INVALID_FORMAT"
    finally: | is-exception _ |
      if is-exception:
        logger_.error "firmware update: failed to apply patch" --tags={"offset": reader.processed}
```

Reply:
done.